auth/src/main/kotlin/com/hedvig/android/auth/AuthRepository.kt

```diff
@@ -67,6 +71,13 @@ sealed interface LoginStatusResult {
 
 fun LoginStatusResult.isPending() = this is LoginStatusResult.Pending
 
+
+sealed interface SubmitOtpResult {
+  data class Error(val message: String) : SubmitOtpResult
+  data class Success(val loginAuthorizationCode: LoginAuthorizationCode) : SubmitOtpResult
+}
```

Comment:
Feel free to not do this if you don't want to, but something I sometimes like in these cases is instead to have
```kotlin
data class SubmitOtpError(val message: String)
data class SubmitOtpSuccess(val loginAuthorizationCode: LoginAuthorizationCode)
```
And have the function return `Either<SubmitOtpError, SubmitOtpSuccess>` instead.
This is so that then on the call site you got the full power of Either to do stuff instead of having to manually deal with the custom result type.
Along with being able to construct this return type by using the `either {}` api like we do in [here](https://github.com/HedvigInsurance/android/blob/1a49ac6baed33b03680f6ae889ad337ccdbff775/app/src/main/kotlin/com/hedvig/app/feature/offer/OfferRepository.kt#L39-L58) for example. So gives some goodies on both receiving that type, and creating it.

It does of course depend on how we want to use this, maybe sometimes it doesn't make sense, but just saying :D

Reply:
Agree yes!